content/posts/nats-multi-tenant/index.md

```diff
@@ -0,0 +1,295 @@
+can create *consumer queue* by `$JS.API.CONSUMER.*.student.*` and `$JS.API.CONSUMER.DURABLE.CREATE.student.*`, then 
+ack messages by `$JS.ACK.student.>`.
+
+#### **Let's make a test**
+
+You can go to [my demo](https://github.com/manabie-com/manabie-com.github.io/tree/setup-nats-multi-tenant/content/posts/nats-multi-tenant/examples) 
+for more detail.
+
+##### **Setup config and docker-compose**
+
+In this demo I just configure one account and three users, you can custom the test with more than accounts. We run command 
+`docker-compose up` to start NATS server, the logs when start NATS successfully.
+
+```bash
+n1    | [1] 2021/12/14 08:12:08.709249 [INF] Starting nats-server
+n1    | [1] 2021/12/14 08:12:08.709303 [INF]   Version:  2.6.6
+n1    | [1] 2021/12/14 08:12:08.709306 [INF]   Git:      [878afad]
+n1    | [1] 2021/12/14 08:12:08.709310 [INF]   Name:     NBWOQ2DYVAYSY3HXYVJXGGURKJY6TBPC2JCKEQYND7GTSSJODGYUCMLK
+n1    | [1] 2021/12/14 08:12:08.709318 [INF]   Node:     Cq4JgPlu
+n1    | [1] 2021/12/14 08:12:08.709321 [INF]   ID:       NBWOQ2DYVAYSY3HXYVJXGGURKJY6TBPC2JCKEQYND7GTSSJODGYUCMLK
+n1    | [1] 2021/12/14 08:12:08.709325 [WRN] Plaintext passwords detected, use nkeys or bcrypt
+n1    | [1] 2021/12/14 08:12:08.709336 [INF] Using configuration file: /jetstream.config
+n1    | [1] 2021/12/14 08:12:08.710457 [INF] Starting JetStream
+n1    | [1] 2021/12/14 08:12:08.710696 [INF]     _ ___ _____ ___ _____ ___ ___   _   __  __
+n1    | [1] 2021/12/14 08:12:08.710708 [INF]  _ | | __|_   _/ __|_   _| _ \ __| /_\ |  \/  |
+n1    | [1] 2021/12/14 08:12:08.710712 [INF] | || | _|  | | \__ \ | | |   / _| / _ \| |\/| |
+n1    | [1] 2021/12/14 08:12:08.710714 [INF]  \__/|___| |_| |___/ |_| |_|_\___/_/ \_\_|  |_|
+n1    | [1] 2021/12/14 08:12:08.710717 [INF] 
+n1    | [1] 2021/12/14 08:12:08.710720 [INF]          https://docs.nats.io/jetstream
+n1    | [1] 2021/12/14 08:12:08.710722 [INF] 
+n1    | [1] 2021/12/14 08:12:08.710725 [INF] ---------------- JETSTREAM ----------------
+n1    | [1] 2021/12/14 08:12:08.710737 [INF]   Max Memory:      4.00 GB
+n1    | [1] 2021/12/14 08:12:08.710742 [INF]   Max Storage:     10.00 GB
+n1    | [1] 2021/12/14 08:12:08.710745 [INF]   Store Directory: "/data/jetstream"
+```
+
+##### **Golang code**
+* **Publisher** folder try connect to NATS with user *Bob* then checking and creating streams `student` after that publish
+three messages with subject `student.Created`. Go to publisher folder then run command `go run publisher.go`.
+
+```bash
+2021/12/14 15:21:02 Student with StudentID:1 has been published
+2021/12/14 15:21:02 Student with StudentID:2 has been published
+2021/12/14 15:21:02 Student with StudentID:3 has been published
+```
+
+* **Consumer** folder try connect to NATS with user *Tom* then [creating subscription](https://github.com/manabie-com/manabie-com.github.io/blob/a9b1ec1f6b87c3b408516014a06850869c7b30f8/content/posts/nats-multi-tenant/examples/consumer/consumer.go#L29) 
```

Comment:
I think we should run the consumers first, because of the `InterestPolicy`. If we run the publisher first here, then all the messages will be gone beforethe subscriber can get them

Reply:
I will change stream's policy is `LimitsPolicy`, we should run publisher first, its can be easier readers to understand

Reply:
that'll work too, nice!